### ISSUE TITLE
conflict between pyscf and pyscfad's loggers

### ISSUE THREAD

**Issue by mkrompiec** [fishjojo/pyscfad#50]
Verified with pyscf 2.6.2 and 2.7.0, pyscfad 0.17
```
from pyscfad import scf as scf_ad  # comment this out and the error disappears
# note that the following calls only pyscf, not pyscfad
from pyscf import scf, gto, mcscf
mol = gto.Mole()
mol.verbose = 3
b = 1.5
mol.atom = [
    ["N", (0., 0., 0.)],
    ["N", (0., 0., b)],]
mol.basis = "sto-3g"
mol.build()
mf = scf.RHF(mol)
mf.run()
mc = mcscf.CASSCF(mf, ncas=2, nelecas=2)
mc.natorb=True
e1 = mc.mc1step()[0]
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[1], line 15
     13 mc = mcscf.CASSCF(mf, ncas=2, nelecas=2)
     14 mc.natorb=True
---> 15 e1 = mc.mc1step()[0]

File ~/.pyenv/versions/3.12.5/lib/python3.12/site-packages/pyscf/mcscf/mc1step.py:869, in CASSCF.mc1step(self, mo_coeff, ci0, callback)
    868 def mc1step(self, mo_coeff=None, ci0=None, callback=None):
--> 869     return self.kernel(mo_coeff, ci0, callback)

File ~/.pyenv/versions/3.12.5/lib/python3.12/site-packages/pyscf/mcscf/mc1step.py:864, in CASSCF.kernel(self, mo_coeff, ci0, callback, _kern)
    857 self.dump_flags()
    859 self.converged, self.e_tot, self.e_cas, self.ci, \
    860         self.mo_coeff, self.mo_energy = \
    861         _kern(self, mo_coeff,
    862               tol=self.conv_tol, conv_tol_grad=self.conv_tol_grad,
    863               ci0=ci0, callback=callback, verbose=self.verbose)
--> 864 logger.note(self, 'CASSCF energy = %#.15g', self.e_tot)
    865 self._finalize()
    866 return self.e_tot, self.e_cas, self.ci, self.mo_coeff, self.mo_energy

File ~/.pyenv/versions/3.12.5/lib/python3.12/site-packages/pyscf/lib/logger.py:140, in note(rec, msg, *args)
    138 def note(rec, msg, *args):
    139     if rec.verbose >= NOTICE:
--> 140         flush(rec, msg, *args)

File ~/.pyenv/versions/3.12.5/lib/python3.12/site-packages/pyscfad/lib/logger.py:47, in flush(rec, msg, *args)
     45     jax.debug.callback(_flush, *args)
     46 else:
---> 47     _flush(*args)

File ~/.pyenv/versions/3.12.5/lib/python3.12/site-packages/pyscfad/lib/logger.py:40, in flush.<locals>._flush(*args)
     39 def _flush(*args):
---> 40     rec.stdout.write(msg % args)
     41     rec.stdout.write('\n')
     42     rec.stdout.flush()

TypeError: not enough arguments for format string


```

**Comment by mkrompiec** [comment]
While the fix to logger.flush will make the code work properly, it is still rather concerning that pyscfad's logger is called by pyscf's CASSCF, isn't it?

**Comment by fishjojo** [comment]
pyscfad calls pyscf functions sometimes with jax tracer objects. Without overriding the pyscf logger, many functions will have to be redefined and duplicated. A better solution is to update the logger inside pyscf, which may happen in the future.

**Comment by mkrompiec** [comment]
I am still seeing this issue, in a slightly different context:
```
from pyscfad import scf as scf_ad  # comment this out and the error disappears
# note that the following calls only pyscf, not pyscfad
from pyscf import scf, gto, cc
mol = gto.Mole()
mol.verbose = 3
b = 1.5
mol.atom = [
    ["N", (0., 0., 0.)],
    ["N", (0., 0., b)],]
mol.basis = "sto-3g"
mol.build()
mf = scf.RHF(mol)
mf.run()
mycc=cc.UCCSD(mf)
mycc.verbose=4  # verbose=3 is fine
mycc.run()
```


```
File ~/.pyenv/versions/3.12.7/envs/main_env/lib/python3.12/site-packages/pyscf/cc/ccsd.py:1025, in CCSDBase.dump_flags(self, verbose)
   1023 log.info('******** %s ********', self.__class__)
   1024 log.info('CC2 = %g', self.cc2)
-> 1025 log.info('CCSD nocc = %s, nmo = %s', self.nocc, self.nmo)
   1026 if self.frozen is not None:
   1027     log.info('frozen orbitals %s', self.frozen)

File ~/.pyenv/versions/3.12.7/envs/main_env/lib/python3.12/site-packages/pyscf/lib/logger.py:136, in info(rec, msg, *args)
    134 def info(rec, msg, *args):
    135     if rec.verbose >= INFO:
--> 136         flush(rec, msg, *args)

File ~/.pyenv/versions/3.12.7/envs/main_env/lib/python3.12/site-packages/pyscfad/lib/logger.py:37, in flush(rec, msg, *args)
     36 def flush(rec, msg, *args):
---> 37     msg, args = _partial_eval_msg(msg, args)
     39     def _flush(*args):
     40         rec.stdout.write(msg % args)

File ~/.pyenv/versions/3.12.7/envs/main_env/lib/python3.12/site-packages/pyscfad/lib/logger.py:24, in _partial_eval_msg(msg, args)
     22 arg = remaining_args.pop(0)
     23 if not util.is_tracer(arg):
---> 24     partially_evaluated_msg += spec % arg
     25 else:
     26     partially_evaluated_msg += spec

TypeError: not all arguments converted during string formatting
```

**Comment by fishjojo** [comment]
Should be fixed in [0342d5f](https://github.com/fishjojo/pyscfad/commit/0342d5f7cdef37167b7cc48925458bd71263418f).

**Comment by mkrompiec** [comment]
thanks!
